mongo/options/clientoptions.go

```diff
@@ -104,6 +105,7 @@ type ClientOptions struct {
 	DisableOCSPEndpointCheck *bool
 	HeartbeatInterval        *time.Duration
 	Hosts                    []string
+	HTTPClient               *http.Client
```

Comment:
Optional: Consider adding a comment that describes what the `HTTPClient` configuration is used for.

E.g.
```go
// HTTPClient allows specifying a custom HTTP client used to call the OCSP TLS certificate
// revocation endpoint specified by TLS certificates. If nil, then the Go Driver uses its
// default HTTP client (internal.DefaultHTTPClient).
```